### PR TITLE
scooter: init at 0.1.1

### PR DESCRIPTION
**Pull request by felixzieger**
Hello, I would like to package scooter: https://github.com/thomasschafer/scooter

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7141,6 +7141,10 @@
     githubId = 628359;
     name = "Felix Singer";
   };
+  felixzieger = {
+    name = "Felix Zieger";
+    github = "felixzieger";
```
Comment (nyabinary) [inline maintainers/maintainer-list.nix:7146]:
```suggestion
    github = "felixzieger";
    githubId = 67903933;
```
https://api.github.com/users/felixzieger

Please add email and matrix if possible as well.

File: pkgs/by-name/sc/scooter/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "thomasschafer";
+    repo = "scooter";
+    rev = "v${version}";
+    hash = "sha256-5YosLJHfrXv6ev7r0gAW5bzkSIwIBUj/Fl3+tklTKQo=";
+  };
+
+  cargoHash = "sha256-GY52JN07igDJVuKXR9iIqNj1obhPs70dOmU1wGy8ZrI";
+
+  meta = with lib; {
```
Comment (nyabinary) [inline pkgs/by-name/sc/scooter/package.nix:16]:
```suggestion
  meta = {
```
See https://github.com/NixOS/nixpkgs/issues/208242 https://github.com/NixOS/nixpkgs/pull/293767

File: pkgs/by-name/sc/scooter/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    description = "Interactive find and replace in the terminal";
+    homepage = "https://github.com/thomasschafer/scooter";
+    changelog = "https://github.com/thomasschafer/scooter/commits/v${version}";
+    license = with licenses; [ mit ];
```
Comment (nyabinary) [inline pkgs/by-name/sc/scooter/package.nix:20]:
```suggestion
    license = lib.licenses.mit;
    platforms = lib.platforms.unix ++ lib.platforms.windows;
```
See https://github.com/NixOS/nixpkgs/issues/208242 https://github.com/NixOS/nixpkgs/pull/293767

File: pkgs/by-name/sc/scooter/package.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    maintainers = with maintainers; [ felixzieger ];
```
Comment (nyabinary) [inline pkgs/by-name/sc/scooter/package.nix:21]:
```suggestion
    maintainers = with lib.maintainers; [ felixzieger ];
```
See https://github.com/NixOS/nixpkgs/issues/208242 https://github.com/NixOS/nixpkgs/pull/293767

**Comment by Gliczy** [comment]
Welcome to nixpkgs!

One CI check is currently failing because your package is not properly formatted.
You must run `nixfmt` on your file to properly format it (you can get `nixfmt` from nixpkgs or simply by running `nix-shell` inside your cloned repo).

Also, the title of this PR should be the same as your commit message "scooter: init at 0.1.1".

**Review by nyabinary — CHANGES_REQUESTED** [review]
Please format the package.nix with ``nixfmt-rfc-style`` to pass the formatting check.

Also, as the comment above stated, please rename the title of this PR to ``scooter: init at 0.1.1``

**Review by UnsolvedCypher — APPROVED** [review]

**Comment by felixzieger** [comment]
> Hi all, hoping I might be able to get some pointers on this - it looks like there were [a couple of automated PRs](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+scooter) to bump the version of scooter but now it appears to be stuck on 0.2.1, when the latest version is 0.2.3. (While 0.2.3 is only a few days old, 0.2.2 was released a few weeks ago.) Any ideas about what could be causing this? I'm not very familiar with Nix so I'd really appreciate any pointers 🙏

Hi @thomasschafer, thanks for checking what is going on with the updates.

The automated package updates are created by [nixpkgs-update](https://nix-community.github.io/nixpkgs-update/#introduction). The logs for scooter updates of nixpkgs-update are available [here](https://nixpkgs-update-logs.nix-community.org/scooter/).

For the 0.2.1 -> 0.2.2 update, we have a [failing test](https://nixpkgs-update-logs.nix-community.org/scooter/2025-02-02.log).

Do you have an idea why this test might have failed?

**Comment by thomasschafer** [comment]
> Hi @thomasschafer, thanks for checking what is going on with the updates.
> 
> The automated package updates are created by [nixpkgs-update](https://nix-community.github.io/nixpkgs-update/#introduction). The logs for scooter updates of nixpkgs-update are available [here](https://nixpkgs-update-logs.nix-community.org/scooter/).
> 
> For the 0.2.1 -> 0.2.2 update, we have a [failing test](https://nixpkgs-update-logs.nix-community.org/scooter/2025-02-02.log).
> 
> Do you have an idea why this test might have failed?

Thank you @felixzieger, this is really helpful!

I'm struggling to work out what could be causing this failure, as tests are running fine for me both locally and in CI on Ubuntu, Windows and macOS, so I'm trying to work out how to reproduce this locally - would my best bet be to run nixpkgs-update in a nix container/VM (given that I use macOS locally)?

**Comment by felixzieger** [comment]
I managed to reproduce it on one of my hosts (I tried to [update to 0.2.3](https://github.com/NixOS/nixpkgs/commit/0e9f0ac6b0cea63850cd65601199fa337b637995)), but I do not understand [buildRustPackage](https://ryantm.github.io/nixpkgs/languages-frameworks/rust/#running-package-tests) well enough to figure out what is going on.

I found an option to disable the failing test. I am in favor of going with this for now and opened a PR for this here: https://github.com/NixOS/nixpkgs/pull/381471

**Comment by thomasschafer** [comment]
> I managed to reproduce it on one of my hosts, but I do not understand [buildRustPackage](https://ryantm.github.io/nixpkgs/languages-frameworks/rust/#running-package-tests) well enough to figure out what is going on.
> 
> I found an option to disable the failing test. I am in favor of going with this for now and opened a PR for this here: #381471

Ah brilliant, thanks so much for this! 🙌 I was going to suggest skipping the test, appreciate you putting up the PR

**Pull request merged**